src/ui.rs

```diff
@@ -40,8 +48,23 @@ pub enum Action {
     Noop,
 }
 
+/// Modal setting for Program, this dictates the commands that are available to the user
+#[derive(PartialEq, Clone)]
+pub enum Mode {
+    /// Default mode, allows the removal, traversal, move, and copy of images
+    Normal,
+    /// Mode that is built off of user input, allows switching the current glob
+    /// string is the input to display on the infobar
+    Command(String),
+    /// Mode that is meant to display errors to the user through the infobar
+    /// string is the input to display on the infobar
+    Error(String),
+    /// Terminate condition, if this mode is set the program will stop execution
+    Exit,
+}
+
 /// State tracks events that will change the behaviour of future events. Such as key modifiers.
-pub struct State {
+pub struct State<'a> {
     /// left_shift tracks whether or not the left shift key is pressed.
```

Comment:
We can probably get rid of tracking `left_shift` and `right_shift` now that textinput will give the character ex. "G"